# DNA Databases

<img align="left" border="10" style="border-color:white" src="images/NCBI.png"> At the advent of DNA sequencing, in the 1980’s, there were no databases, but there were also very few scientific journals. As people started generating sequences, a few researchers petitioned the editors of the scientific journals to create a standard: prior to publishing the sequence of a piece of DNA, the authors must deposit the sequence in a publicly available database. This ensured that all the sequences that had been generated were available for other scientists to use. 
<img align="right" border="10" style="border-color:white" src="images/EMBL.png"> The first three databases became the National Center for Biotechnology Information (NCBI; ), the DNA Database of Japan (DDBJ), and the European Bioinformatics Institute (EBI). To ensure rapid access of all sequences to all researchers, these three databases agreed to share their DNA sequences nightly. You can submit a DNA sequence to any of these three databases, and it will appear in the other two when they synchronize.

<img align="left" border="10" style="border-color:white" src="images/DDBJ.png"> One of the problems with this approach, however, was that different research groups sequenced the same thing. In the early days of DNA sequencing competing scientists working on the same gene would sequence it; sometimes people sequenced the same organism; and of course, you will find the same protein in different organisms. To overcome this redundancy in the data, the sequence databases introduced the concept of “**non-redundant**” databases. These databases only have one version of each sequence, and from that version you can access the different sources of the sequence. NCBI made two different non-redundant databases, one called **nr** for proteins, and one called **nt** for nucleotides.

<img align="right" border="10" style="border-color:white" src="images/UniProt.png"> As genome sequencing grew in popularity, several parallel efforts were started to organize the genomic information. One of the first efforts was **Swiss-Prot**, started in 1986 by Amos Bairoch. A similar effort, the **Protein Information Resource** was started by Margaret Dayhoff (who invented PAM matrices for protein alignment), and after her death was run by Cathy Wu. Swiss-Prot and PIR joined forces in 2003 to form **UniProt**. The **KEGG** (Kyoto Encyclopedia of Genes and Genomes; http://www.genome.jp/kegg/) effort was started in 1995 as a spin off from the Japanese human genome effort. Each of these resources has different takes and views on the same data, so when you are interrogating a sequence it is usually wise to view all the resources to see how they have interpreted it.

<img align="left" border="10" style="border-color:white" src="images/JGI.png">The Department of Energy’s [Joint Genome Institute](http://jgi.doe.gov/) (JGI) initially focused on sequencing the human genome, but also moved into plant and microbial sequencing. They required a database to house the genome sequences they were producing, and started the [Integrated Microbial Genomes](https://img.jgi.doe.gov/) (IMG) effort to curate the genomes they were sequencing. 
<img align="right" border="10" style="border-color:white" src="images/seed.png">The SEED group grew out of work that started with Carl Woese, Ross Overbeek, and Gary Olsen sequencing and annotating archaeal genomes ([Bult et al., 1996](https://www.ncbi.nlm.nih.gov/pubmed/8688087); [Klenk et al., 1997](https://www.ncbi.nlm.nih.gov/pubmed/9389475)), and after several iterations through WIT (What is there?) and ERGO (Overbeek et al., 2003), the Fellowship for the Interpretation of Genomes (FIG) started first the SEED, and later the SAPLING databases (Overbeek, Disz & Stevens, 2004; Overbeek et al., 2005). 

The major problem that these groups were trying to tackle was that because all of the sequencing and annotation was done by different groups, each group used its own nomenclature for the same thing. For example, the genes involved in regulation of nitrogen fixation were called glnL and glnG by some groups (because they regulated glutamine synthetase) and ntrB and ntrC by other groups (because they regulated nitrogen utilization) (Merrick & Edwards, 1995). One approach to normalize these annotations is to group all the related proteins into a bin, and then give that bin a consistent name. The bins are sometimes called orthologous groups (e.g. KO from KEGG; COG from NCBI), sometimes they are called families (e.g. TIGRfam from the (now defunct) The Institute for Genomic Research (TIGR); FIGfam from FIG), etc. 
